WordPress/gutenberg
# Distraction free mode: Fix keyboard shortcut not working

Fixes: #47873
Related to: #43428

## What?
This PR fixes a problem where the shortcut for switching distraction flow mode did not work (probably Windows only).

## Why?

The shortcut for switching the distraction free mode is as follows:

- Windows: `Ctrl`+`Shift`+`\`
- Mac: `⇧`+`⌘`+`\`

The `event.key` for determining shortcuts has a value that takes into account modifier keys such as the `shift` key. Therefore, as I have investigated in this comment, I believe the cause is that when the `shift` key is combined with other keys, `event.key` has a value that is not originally expected. Furthermore, the character changed by the `shift` key depends on the keyboard layout, so `event.key` cannot be used as the source of the decision.

## How?

Refers to the value of `event.code` that represents a physical key on the keyboard. Based on that value, it is converted to the originally expected `event.key`.

This PR fixes the bug in Windows. However, as mentioned in [the two comments from here](https://github.com/WordPress/gutenberg/issues/47873#issuecomment-1423175792), the test results seem to be different for Mac. I would appreciate it if you could test to see if this shortcut issue is currently occurring on Mac.

### Testing Instructions for Keyboard

- Confirm that this shortcut key works in the post editor.
- Confirm that the other shortcuts also work as before.

## Screenshots or screencast

### English Keyboard Layout

`event.code` is `Backslash`.

https://user-images.githubusercontent.com/54422211/217984887-e7fb7f37-2242-4b7e-989e-a2cade0e56be.mp4

### Japanese Keyboard Layout

`event.code` is `IntlRo` or `IntlYen`.

https://user-images.githubusercontent.com/54422211/217984898-dcf49982-636d-430c-ae33-ab28575f4d03.mp4
